include/functions_entries.inc.php

```diff
@@ -146,10 +146,15 @@ function &serendipity_fetchEntryCategories($entryid) {
                 ORDER BY c.category_name ASC";
 
         $cat =& serendipity_db_query($query);
+        
         if (!is_array($cat)) {
             $arr = array();
             return $arr;
         } else {
+            foreach ($cat as $key => $catdata) {
```

Comment:
This is not very good performance-wise to call it for each key individually. Please move this hook outside the foreach loop to operate on the whole $cat array?

Reply:
The functionality of the hook is to remove the translation tags. If there is a result array passed via the hook, is it safe to purge the tags from all the result fields? That was the question, because the issuing queries are different. Or I have to pull the values to be stripped out of the array in a new one, put that into the hook and reinsert the result. Seems doable.

Reply:
Yes, you can pass $cat as $eventData to the hook. It's passed by reference, so any modification the plugins make then to $eventData on the hook will continue in the loop of functions_entries.inc.php to hold that modified data. Then only your plugin needs to make the foreach loop on $eventData, and the whole plugin API introspection is only called once instead of X times.

Reply:
What I meant is that it is not so simple. $cat may be a multidimensional array and then I'd need to expand the tag removing function on the other side that it iterates through the whole array. But that may have unintended consequences because I only want to remove the tags from the 'category name' and 'category description' fields and from the whole query result. the tags may be quite unique, but who knows … And because the queries are not consistent, the fields can also be numbered ('assoc' option or not). To be safe, I have to copy the fields to strip out of the result array, pass them to the hook and reinsert them afterwards. A little more code, and because of that it was easier to iterate inside the source function and do the multiple calls to the hook. But as I said, I can do that (maybe this evening).

Reply:
You could do the distinction on the event hook itself? It knows the context and what is passed, so you could pass 'addData' along to specify to the plugin that it needs in which context the plugin hook was called. Another way is to not use one central hook, but use one hook for multi-dimensional arrays and one for the other?